cmd/livepeer/livepeer.go

```diff
@@ -901,6 +904,14 @@ func main() {
 		drivers.NodeStorage = drivers.NewMemoryDriver(n.GetServiceURI())
 	}
 
+	if *metadataAmqpUri != "" {
+		uri, exchange, keyNs := *metadataAmqpUri, *metadataExchange, n.NodeType.String()
```

Comment:
`n.NodeType.String()` is going to be something like "broadcaster", "orchestrator", etc. (since `n.NodeType.String()` is a [enum](https://github.com/livepeer/go-livepeer/blob/4e24711bfabc873f1be7e98cdd8599f68a6cfedd/core/livepeernode.go#L38)). IIUC this ends up being the `keyNs` arg when creating the AMQP producer which is a prefix for all keys used in messages. Do we want all keys to be prefixed with strings in this manner and if so, what is the motivation for this?

Reply:
Yeah! I explicitly added that `String()` implementation to provide a stable namespacing for the keys. The whole logic for how I designed the AMQP topics is described in full here: https://github.com/livepeer/go-livepeer/pull/1946#discussion_r668326074

The TL;DR for this specific section is that AMQP allows for subscribing (binding) to specific exchanges under specific topics, and there are some handy "wildcards" that can be used for one or many segments in the topic (aka routing key). 

So for example, all events from the broadcaster would be like `broadcater.stream_health.transcode.<shardKey>.<streamId>`. An orchestrator could send the same event, but it'd have an `orchestrator.` prefix instead.

A consumer, on the other side, could independently subscribe to only `broadcaster` or `orchestrator` events, or if they don't care about the node type just subscribe to `*.stream_health.transcode...`. The segments in the topic serve to _allow_ consumers to be more specific in the kinds of events that they want to receive, but don't oblige them to.

So I thought the node type could be useful at some point, but if you think it's unlikely, we can also use something else or just use nothing at all as well. I wanted to get the region in there somehow but I think it's not available to the code here, and thinking further about it maybe it doesn't make much sense anyway.